pkg/controller/security/security_manager.go

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package security
+
+import (
+	"sync"
+	"time"
+
+	"container/heap"
+
+	istiosecurity "istio.io/istio/pkg/security"
+	"istio.io/istio/pkg/spiffe"
+	"kmesh.net/kmesh/pkg/controller/workload"
+	"kmesh.net/kmesh/pkg/logger"
+)
+	
+type Cert struct {
+	hostName string
+	exp time.Time
+}
+
+type PriorityQueue []*Cert
+
+var certsMaps sync.Map
+var saSaveUidNum = make(map[string]int)
+var log = logger.NewLoggerField("security")
+
+type SecretManagerClient struct {
+	caClient *KmeshCaClient
+
+	// configOptions includes all configurable params for the cache.
+	configOptions *istiosecurity.Options
+
+	// storing certificates
+	secretcache *sync.Map
+
+	//caRootPath string
+	pending	PriorityQueue
+}
+ 
+var ScClient SecretManagerClient
```

Comment:
It is better to creat an instance of `SecretManagerClient` and encapsulate it into other data structure instead of using global variables directly. This make it easier to manage.